### DISCUSSION TITLE
template erases input content on update

### DISCUSSION THREAD

**Discussion by anjanesh** [alpinejs/alpine#1018] [1. Help] [answered]
```
<script>const one_contact = { name: '' };</script>
<div x-data="{ persons: [] }">
    <template x-for="person in persons">
        <div><input type="text" name="names[]" x-bind:value="person.name" /></div>
    </template>            
    <button x-on:click="persons = persons.concat(one_contact)">Add Person</button>
</div>
```
On entering data in the input textbox and then clicking "Add Person", erases the content of all the entered textboxes.

**Comment by KevinBatdorf** [comment, marked as answer]
Try this:
```html
<script>const one_contact = { name: '' };</script>
<div x-data="{ persons: [] }">
    <template x-for="(_, i) in persons">
        <div><input class="border" type="text" x-model="persons[i].name" /></div>
    </template>            
    <button x-on:click="persons.push(Object.assign({}, one_contact))">Add Person</button>
</div>
```
Demo: https://codepen.io/KevinBatdorf/pen/vYXvPeG

You should consider adding a unique ID to the `one_contact` object though and use it on the `x-for` as the `:key` binding. A simple id generator is: `id: Date.now() + Math.floor(Math.random() * 1000000)`

**Reply by anjanesh** [reply]
Thank you for your response. I have updated my code :
```
<div x-data="{ persons: [] }">
    <template x-for="(person , i) in persons" :key="persons[i].id">
        <div><input type="text" name="names[]" x-model="persons[i].name" /></div>
    </template>            
    <button x-on:click="persons.push(Object.assign({}, { name: '', id: Date.now() + Math.floor(Math.random() * 1000000) }))">Add Person</button>
</div>
```
But why do we have to do `persons.push(Object.assign({}, one_contact))` rather than `persons.push(one_contact)` ?

**Reply by KevinBatdorf** [reply]
When you declare `const one_contact = { name: '' }` `one_contact` is a reference to the object. `Object.assign()` will create a new object and pass in the parameters from `one_contact`.

**Reply by anjanesh** [reply]
In that case if we push an object directly to persons.push, it works.
`<button x-on:click="persons.push({ name: '' })">Add Person</button>`
